src/inference/inference_tensorflowlite.py

```diff
@@ -0,0 +1,299 @@
+    else:
+        raise ValueError(f'Unsupported file format of the model: {suffix}')
+
+
+def inference_tflite(interpreter, number_iter, get_slice):
+    result = None
+    time_infer = []
+    interpreter.allocate_tensors()
+    model_inputs = interpreter.get_input_details()
+    input_info = {}
+    for model_input in model_inputs:
+        input_info[model_input['name']] = (model_input['index'], model_input['dtype'], model_input['shape'])
+
+    outputs = interpreter.get_output_details()
+    for i in range(number_iter):
+        for name, data in get_slice(i).items():
+            interpreter.set_tensor(input_info[name][0], data.astype(input_info[name][1]))
+        interpreter.invoke()
+        t0 = time()
+        result = [interpreter.get_tensor(output['index']) for output in outputs]
+        t1 = time()
+        time_infer.append(t1 - t0)
+
+    return result, time_infer
+
+
+def reshape_model_input(io_model_wrapper, model, log):
+    model_inputs = model.get_input_details()
+    for model_input in model_inputs:
+        shape = io_model_wrapper.get_input_layer_shape(model, model_input['name'])
+        if (shape != model_input['shape']).any():
+            log.info('Reshaping model input from {0} to {1}'.format(model_input['shape'], shape))
+            model.resize_tensor_input(model_input['index'], shape)
+
+
+def process_result(batch_size, inference_time):
+    inference_time = pp.three_sigma_rule(inference_time)
+    average_time = pp.calculate_average_time(inference_time)
+    latency = pp.calculate_latency(inference_time)
+    fps = pp.calculate_fps(batch_size, latency)
+
+    return average_time, latency, fps
+
+
+def result_output(average_time, fps, latency, log):
+    log.info('Average time of single pass : {0:.3f}'.format(average_time))
+    log.info('FPS : {0:.3f}'.format(fps))
+    log.info('Latency : {0:.3f}'.format(latency))
+
+
+def raw_result_output(average_time, fps, latency):
+    print('{0:.3f},{1:.3f},{2:.3f}'.format(average_time, fps, latency))
+
+
+def create_dict_for_transformer(args):
+    dictionary = {}
+    for i, name in enumerate(args.input_name):
+        channel_swap = args.channel_swap[i] if i < len(args.channel_swap) else None
+        mean = args.mean[i] if i < len(args.mean) else None
+        input_scale = args.input_scale[i] if i < len(args.input_scale) else None
+        layout = args.layout[i] if i < len(args.layout) else None
+        dictionary[name] = {'channel_swap': channel_swap, 'mean': mean,
+                            'input_scale': input_scale, 'layout': layout}
+
+    return dictionary
+
+
+def main():
+    log.basicConfig(format='[ %(levelname)s ] %(message)s',
+                    level=log.INFO, stream=sys.stdout)
+    args = cli_argument_parser()
+    try:
+        model_wrapper = TensorFlowLiteIOModelWrapper(args)
+
+        delegate = None
+        if args.delegate_ext:
+            log.info('Loading delegate library from {0}'.format(args.delegate_ext))
```

Comment:
please use f-strings here and elsewhere. Two types of string format are not desirable if there is no need to change the format of the number

Reply:
Done